dask/blockwise.py

```diff
@@ -1401,6 +1411,10 @@ def rewrite_blockwise(inputs):
                     indices.append(index)
             new_dsk = subs(inputs[dep].dsk, sub)
 
+            # Change new_dsk key to match local_dep
```

Comment:
>Just still wondering about this. Can we be guaranteed that the dsk contains no references to the key dep? I just don't know.

I'm definitely not sure I understand what you are asking here. Maybe you could share a hypothetical graph with the problem you are thinking of? 

>In practice I think the graphs are totally formulaic, but in principle couldn't the dsk of a Blockwise be any arbitrary valid graph?

I am pretty sure the answer is no.  At this point `dsk` is not a SubgraphCallable yet, and so it's guaranteed to be pretty simple (i.e. `{<layer-name>: (<function>, <one-or-more-blockwise-tokens>)}`). 

>Would we want to do this renaming via subs or something just to be sure we're catching everything? In the same way we use subs because we're renaming the blockwise tokens?

When we rename blockwise tokens, we are modifying the values (and not the keys) in the dictionary.  In this case, we are only changing a single key. I am not really understanding what would be missed here.

Reply:
Ah - Okay.  Thanks for clarifying your thoughts here. I think I agree with everything you said :)